### PR TITLE
AMP-48899 Updated Kotlin codegen: added explicit parameter names/types to resolve resolution ambiguity

### PR DESCRIPTION
**Pull request by falconandy**
Added both argument names and types

### CODE REVIEW

File: android/kotlin/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.kt [hunk 1]
```diff
@@ -77,10 +77,13 @@ class Identify private constructor(
...
-    ) : this(mapOf(
-        *(if (optionalArray != null) arrayOf("optionalArray" to optionalArray) else arrayOf()),
-        "requiredNumber" to requiredNumber
-    ))
+    ) : this(
+        eventProperties = mapOf(
+            *(if (optionalArray != null) arrayOf("optionalArray" to optionalArray) else arrayOf()),
+            "requiredNumber" to requiredNumber
+        ) as Map<String, Any?>?,
+        options = null as EventOptions?
```
Comment (qingzhuozhen) [inline android/kotlin/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.kt:85]: Do we need this options null here?

Comment (justin-fiedler) [reply android/kotlin/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.kt:85]: +1 https://github.com/amplitude/javascript/pull/14823#discussion_r809263556

Comment (falconandy) [reply android/kotlin/AmpliApp/app/src/main/java/com/amplitude/ampli/Ampli.kt]: Fixed

**Review by qingzhuozhen — APPROVED** [review]
Is

**Review by justin-fiedler — APPROVED** [review]

**Pull request merged**